driver/ni/ai_channels.h

```diff
@@ -0,0 +1,1316 @@
+                      excitationConfig(parser),
+                      r0(parser.required<double>("r0")) {
+                        std::string u = parser.optional<std::string>("units", "Amps");
+                        this->units = ni::UNITS_MAP.at(u); 
+                      }
+    
+            int32 createNIChannel() override {
+                return ni::NiDAQmxInterface::CreateAIRTDChan(
+                        this->task_handle,
+                        this->name.c_str(),
+                        "",
+                        this->min_val,
+                        this->max_val,
+                        this->units,
+                        this->rtdType,
+                        this->resistanceConfig,
+                        this->excitationConfig.voltageExcitSource, //TODO change name to current
+                        this->excitationConfig.voltageExcitVal, //TODO change name to current
+                        this->r0
+                );
+            }
+    };
+
+///////////////////////////////////////////////////////////////////////////////////
+//                                      Temperature                              //
+///////////////////////////////////////////////////////////////////////////////////
+class Thermocouple : public Analog {
+public:
+    int32_t thermocoupleType;
+    int32_t cjcSource;
+    double cjcVal;
+    std::string cjcChannel;
+
+    static int32_t getType(std::string type) {
+        if (type == "J") return DAQmx_Val_J_Type_TC;
+        if (type == "K") return DAQmx_Val_K_Type_TC;
+        if (type == "N") return DAQmx_Val_N_Type_TC;
+        if (type == "R") return DAQmx_Val_R_Type_TC;
+        if (type == "S") return DAQmx_Val_S_Type_TC;
+        if (type == "T") return DAQmx_Val_T_Type_TC;
+        if (type == "B") return DAQmx_Val_B_Type_TC;
+        if (type == "E") return DAQmx_Val_E_Type_TC;
+
+        LOG(ERROR) << "Invalid TC Type";
+        return DAQmx_Val_J_Type_TC;
+    }
+
+    static int32_t getCJCSource(std::string source) {
+        if (source == "BuiltIn") return DAQmx_Val_BuiltIn;
+        if (source == "ConstVal") return DAQmx_Val_ConstVal;
+        if (source == "Chan") return DAQmx_Val_Chan;
+        LOG(ERROR) << "Invalid cjc type";
+        return DAQmx_Val_BuiltIn;
+    }
+
+
+    explicit Thermocouple(config::Parser &parser, TaskHandle task_handle,
+                          std::string name)
+        : Analog(parser, task_handle, name),
+          thermocoupleType(getType(parser.required<std::string>("thermocouple_type"))),
+          cjcSource(getCJCSource(parser.required<std::string>("cjc_source"))),
+          cjcVal(parser.required<double>("cjc_val")) {
+        std::string u = parser.optional<std::string>("units", "DegC");
+        this->units = ni::UNITS_MAP.at(u); // TODO: make this optional and default to C?
+    }
+
+    //cjcChannel(parser.required<std::string>("cjc_channel")) {} FIXME: this property should be take form console
```

Comment:
Didn't we address this in the form

Reply:
if CJC source is set to "chan", then we expect an input is expected which is an ni channel from which the tc channel can reference for CJC. Right now, the console doesn't provide a way to pass this channel in when that specific cjc source is set,